src/main/java/org/candlepin/async/tasks/CloudAccountOrgSetupJob.java

```diff
@@ -0,0 +1,186 @@
+import org.candlepin.async.JobExecutionContext;
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.service.CloudAccountData;
+import org.candlepin.service.CloudProvider;
+import org.candlepin.service.CloudRegistrationAdapter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+public class CloudAccountOrgSetupJob implements AsyncJob {
+
+    private static final Logger log = LoggerFactory.getLogger(CloudAccountOrgSetupJob.class);
+
+    public static final String JOB_KEY = "AnonymousRegisterJob";
+    public static final String JOB_NAME = "Anonymous Register Job";
+
+    protected static final String CLOUD_ACCOUNT_ID = "cloud_account_id";
+    protected static final String OFFERING_ID = "offering_id";
+    protected static final String CLOUD_PROVIDER = "cloud_provider";
+
+    private final CloudRegistrationAdapter cloudAdapter;
+
+
+    @Inject
+    public CloudAccountOrgSetupJob(CloudRegistrationAdapter cloudAdapter) {
+
+        this.cloudAdapter = Objects.requireNonNull(cloudAdapter);
+    }
+
+    /**
+     * Executes this job.
+     *
+     * @param context the execution context for the job, containing the arguments
+     *                and other context-specific data
+     * @throws JobExecutionException if an exception occurs during job execution
+     */
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        JobArguments args = context.getJobArguments();
+
+        String accountId = args.getAsString(CLOUD_ACCOUNT_ID);
+        String offeringId = args.getAsString(OFFERING_ID);
+        CloudProvider cloudProviderShortName = args.getAs(CLOUD_PROVIDER, CloudProvider.class);
+
+        try {
+            CloudAccountData accountData = this.cloudAdapter.setupCloudAccountOrg(
+                accountId, offeringId, cloudProviderShortName);
+
+            context.setJobResult("Entitled offering %s to owner %s (anonymous: %s).", offeringId,
+                accountData.ownerKey(), accountData.isAnonymous());
+            log.info("Entitled offering {} to owner {} (anonymous: {}).", offeringId,
+                accountData.ownerKey(), accountData.isAnonymous());
+        }
+        catch (Exception e) {
+            throw new JobExecutionException(e.getMessage(), e, false);
+        }
+    }
+
+    /**
+     * Creates a JobConfig configured to execute the isAnonymous register job. Callers may
+     * further manipulate the JobConfig as necessary before queuing it.
+     *
+     * @return a JobConfig instance configured to execute the isAnonymous register job
+     */
+    public static AnonymousRegisterJobConfig createJobConfig() {
+        return new AnonymousRegisterJobConfig();
+    }
+
+    /**
+     * Job configuration object for the export job
+     */
+    public static class AnonymousRegisterJobConfig extends JobConfig<AnonymousRegisterJobConfig> {
```

Comment:
This inner class name should also change as the main class name

Reply:
done